app/javascript/dashboard/store/modules/conversations/actions.js

```diff
@@ -86,6 +86,62 @@ const actions = {
     }
   },
 
+  syncActiveConversationMessages: async (
+    { commit, state, dispatch },
+    { conversationId }
+  ) => {
+    const { allConversations, syncConversationsMessages } = state;
+    const lastMessageId = syncConversationsMessages[conversationId];
+    const selectedChat = allConversations.find(
+      conversation => conversation.id === conversationId
+    );
+    if (!selectedChat) return;
+    try {
+      const { messages } = selectedChat;
+      const {
+        data: { meta, payload },
+      } = await MessageApi.getPreviousMessages({
+        conversationId,
+        after: lastMessageId,
+      });
+      commit(`conversationMetadata/${types.SET_CONVERSATION_METADATA}`, {
+        id: conversationId,
+        data: meta,
+      });
+      const missingMessages = payload.filter(
+        message => !messages.find(item => item.id === message.id)
+      );
+      commit(types.SET_MISSING_MESSAGES, {
```

Comment:
Do we need a separate mutation here? Since we have already filtered the messages previously, is it possible to re-use the mutation that was used before?

Reply:
The current mutation is used for setting all the messages. It can't put messages in a particular order (Missing messages can come at any position.)

Reply:
https://github.com/chatwoot/chatwoot/blob/develop/app/javascript/dashboard/store/modules/conversations/index.js#L54